task/options/options.go

```diff
@@ -40,22 +45,129 @@ type Options struct {
 
 	// Every represents a fixed period to repeat execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Every time.Duration `json:"every,omitempty"`
+	Every Duration `json:"every,omitempty"`
 
 	// Offset represents a delay before execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Offset *time.Duration `json:"offset,omitempty"`
+	Offset *Duration `json:"offset,omitempty"`
 
 	Concurrency *int64 `json:"concurrency,omitempty"`
 
 	Retry *int64 `json:"retry,omitempty"`
 }
 
+type Duration struct {
+	Node ast.DurationLiteral
+}
+
+func (a Duration) String() string {
+	return ast.Format(&a.Node)
+}
+
+func (a *Duration) Parse(s string) error {
+	q, err := parseSignedDuration(s)
+	if err != nil {
+		return err
+	}
+	a.Node = *q
+	return nil
+}
+
+func MustParseDuration(s string) (dur *Duration) {
+	dur = &Duration{}
+	if err := dur.Parse(s); err != nil {
+		panic(err)
+	}
+	return dur
+}
+
+// parseSignedDuration will convert a string into a possibly negative DurationLiteral.
+// TODO(docmerlin): switch to the one in flux
```

Comment:
Why not just use the one you added to Flux?
https://github.com/influxdata/flux/blob/master/parser/strconv.go#L22-L37

Reply:
I think, I am going to switch to using the continue pattern that mark suggested.

Reply:
I agree, that's way more readable. Really nice. I was more referring to performing the check on the actual concrete type vs the string returned by the `Type()` method. I don't expect those methods or the types to be modified. It just seemed like a safer alternative for reducing the chances of unexpected behavior in the future because of an AST change in Flux.